addons/beehave/nodes/beehave_debug.gd

```diff
@@ -0,0 +1,102 @@
+extends BeehaveNode
+
+class_name BeehaveDebug, "../icons/tree.svg"
+
+export(NodePath) var _beehave_tree
+var beehave_tree : BeehaveRoot setget set_beehave_tree
+
+export(Color) var tick_color := Color.whitesmoke
+export(Color) var success_color := Color.forestgreen
+export(Color) var running_color := Color.royalblue
+export(Color) var failure_color := Color.crimson
+
+onready var tree : Tree
+onready var root
+
+var tree_map := {}
+
+func _ready():
+	if self._beehave_tree:
+		self.beehave_tree = get_node(self._beehave_tree)
+	
+func set_beehave_tree(value):
+	if beehave_tree != value:
+		if self.beehave_tree:
+			self.beehave_tree.disconnect("tick_start", self, "_tree_tick_start")
+		
+		beehave_tree = value
+		generate_tree()
+	
+func generate_tree():
+	if self.tree:
+		remove_child(self.tree)
+		
+	for node in self.tree_map.keys():
+		node.disconnect("tick_start", self, "_tick_start")
+		node.disconnect("tick_end", self, "_tick_end")
+	
+	self.tree_map = {}
+	self.tree = Tree.new()
+	self.tree.anchor_left = 0
+	self.tree.anchor_right = 1
+	self.tree.anchor_top = 0
+	self.tree.anchor_bottom = 1
+	self.tree.size_flags_horizontal = self.tree.SIZE_SHRINK_END
+	self.tree.hide_folding = true
+	add_child(self.tree)
+	
+	self.beehave_tree.connect("tick_start", self, "_tree_tick_start")
+	self.root = self.tree.create_item()
+	_add_children(self.root, self.beehave_tree)
+	
+func _add_children(item, node):
+	tree_map[node] = item
+	item.set_text(0, node.name)
+	var icon
+	if node is BeehaveRoot:
+		icon = preload("../icons/tree.svg")
```

Comment:
It might be better to preload these icons `onready` and simply reuse them.

Reply:
You're right. I thought the `preload` would insert the resources inline but it still might be duplicating the resources for every call. I'll have that fixed for a proper PR